### PR TITLE
[solarmax] Support configurable Device Address

### LINKED ISSUE #14365: [Solarmax] Support setting Device Address
In the current implementation, the Solarmax binding always uses the default Device Address (1). The Device Address should be made to be configurable.

### PR DESCRIPTION
**Pull request by jamietownsend**
Closes #14365

Signed-off-by: Jamie Townsend <jamie_townsend@hotmail.com>

### CODE REVIEW

File: bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/i18n/solarmax_it.properties [hunk 1]
```diff
@@ -14,6 +14,8 @@ thing-type.config.solarmax.inverter.host.label = Host
 thing-type.config.solarmax.inverter.host.description = Nome host o indirizzo IP
 thing-type.config.solarmax.inverter.portNumber.label = Porta
 thing-type.config.solarmax.inverter.portNumber.description = Numero porta (valore predefinito 12345)
+thing-type.config.solarmax.inverter.deviceAddress.label = Indirizzo dispositivo
+thing-type.config.solarmax.inverter.deviceAddress.description = Indirizzo dispositivo (defaults to 1)
```
Comment (jlaur) [inline bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/i18n/solarmax_it.properties:18]: This needs to be provided through Crowdin. For reference: https://www.openhab.org/docs/developer/utils/i18n.html#managing-translations

File: bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -37,10 +37,15 @@
...
 			</parameter>
 			<parameter name="portNumber" type="integer" required="false">
-				<label>Port</label>
+				<label>Port Number</label>
 				<description>Port Number (defaults to 12345)</description>
 				<default>12345</default>
 			</parameter>
+			<parameter name="deviceAddress" type="integer" required="false">
+				<label>Device Address</label>
+				<description>Device Address (defaults to 1)</description>
```
Comment (jlaur) [inline bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/thing/thing-types.xml:46]: In the README description is: "Device address for devices connected serially." Consider if you want to bring some of that extended description here, as right now it just duplicates the label and mentions default value.

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Only a small request to remove the Italian translation from this PR and provide it in Crowdin instead after the PR has been merged.

File: bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/thing/thing-types.xml (hunk 2 above)
Comment (jamietownsend) [reply bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/thing/thing-types.xml:46]: Done

File: bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/i18n/solarmax_it.properties (hunk 1 above)
Comment (jamietownsend) [reply bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/i18n/solarmax_it.properties:18]: Deleted

File: bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/i18n/solarmax.properties [hunk 3]
```diff
@@ -12,8 +12,10 @@ thing-type.solarmax.inverter.description = Basic thing for the SolarMax Power In
 
 thing-type.config.solarmax.inverter.host.label = Host
 thing-type.config.solarmax.inverter.host.description = Hostname or IP Address
-thing-type.config.solarmax.inverter.portNumber.label = Port
+thing-type.config.solarmax.inverter.portNumber.label = Port Number
 thing-type.config.solarmax.inverter.portNumber.description = Port Number (defaults to 12345)
+thing-type.config.solarmax.inverter.deviceAddress.label = Device Address
+thing-type.config.solarmax.inverter.deviceAddress.description = Device Address (defaults to 1)
```
Comment (jlaur) [inline bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/i18n/solarmax.properties:18]: Please regenerate, text has now changed to: "Device address for devices connected serially (defaults to 1)"

**Review by jlaur — CHANGES_REQUESTED** [review]
One more comment after last push - also please run `spotless`.

File: bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/i18n/solarmax.properties (hunk 3 above)
Comment (jamietownsend) [reply bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/i18n/solarmax.properties:18]: Ran spotless:apply, committed and pushed.

Comment (jlaur) [reply bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/i18n/solarmax.properties:18]: You also need to regenerate this file, see https://www.openhab.org/docs/developer/utils/i18n.html#generating-i18n-properties-file

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**